pkg/cloudproduct/gke/gke_test.go

```diff
@@ -66,17 +66,201 @@ func TestSyncPodPortsToGameServer(t *testing.T) {
 				return
 			}
 			if assert.NoError(t, err) {
-				if diff := cmp.Diff(tc.wantGS, tc.gs); diff != "" {
-					t.Errorf("GameServer diff (-want +got):\n%s", diff)
-				}
-				if diff := cmp.Diff(oldPod, tc.pod); diff != "" {
-					t.Errorf("Pod was modified (-old +new):\n%s", diff)
-				}
+				require.Equal(t, tc.wantGS, tc.gs)
+				require.Equal(t, oldPod, tc.pod)
 			}
 		})
 	}
 }
 
+func TestValidateGameServer(t *testing.T) {
+	for name, tc := range map[string]struct {
+		gs   *agonesv1.GameServer
```

Comment:
Each test case would be more concise if you just used the slice of `GameServerPorts` instead of the whole struct (the stuff around that part is just boilerplate for these tests). 

Also, are these tests sufficient for all edge cases? (I didn't check closely enough to know, so it's ok to say yes).

Reply:
Agreed on ports. I think the test cases are more than adequate here, and I double checked the coverage layer and it seems to agree - in terms of validation there's not a whole lot. The harder one is actually below, which I'll look at longer.